docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/Gam.cs

```diff
@@ -0,0 +1,153 @@
+            //  Average prediction: 0.82
+            //
+            //  Feature0
+            //  x < -0.44 => 0.286
+            //  x < -0.38 => 0.225
+            //  x < -0.32 => 0.048
+            //  x < -0.26 => -0.110
+            //  x < -0.20 => -0.116
+            //  x < 0.18 => -0.143
+            //  x < 0.25 => -0.115
+            //  x < 0.31 => -0.005
+            //  x < 0.37 => 0.097
+            //  x < 0.44 => 0.263
+            //  x < ∞ => 0.284
+            //
+            //  Feature1
+            //  x < 0.00 => -0.350
+            //  x < 0.24 => 0.875
+            //  x < 0.31 => -0.138
+            //  x < ∞ => -0.188
+
+            // Let's consider this output. To score a given example, we look up the first bin where the inequality
+            // is satisfied for the feature value. We can look at the whole function to get a sense for how the
+            // model responds to the variable on a global level.
+            // The model can be seen to reconstruct the parabolic and step-wise function, shifted with respect to the average
+            // expected output over the training set. Very few bins are used to model the second feature because the GAM model
+            // discards unchanged bins to create smaller models.
+            // One last thing to notice is that these feature functions can be noisy. While we know that Feature1 should be
+            // symmetric, this is not captured in the model. This is due to noise in the data. Common practice is to use 
+            // resampling methods to estimate a confidence interval at each bin. This will help to determine if the effect is 
+            // real or just sampling noise. See for example:
+            // Tan, Caruana, Hooker, and Lou. "Distill-and-Compare: Auditing Black-Box Models Using Transparent Model 
+            // Distillation." <a href='https://arxiv.org/abs/1710.06169'>arXiv:1710.06169</a>."
+        }
+
+        private class Data
+        {
+            public bool Label { get; set; }
+
+            [VectorType(2)]
+            public float[] Features { get; set; }
+        }
+
+        /// <summary>
+        /// Creates a dataset, an IEnumerable of Data objects, for a GAM sample. Feature1 is a parabola centered around 0,
+        /// while Feature2 is a simple piecewise function.
+        /// </summary>
+        /// <param name="numExamples">The number of examples to generate.</param>
+        /// <param name="seed">The seed for the random number generator used to produce data.</param>
+        /// <returns></returns>
+        private static IEnumerable<Data> GenerateData(int numExamples = 25000, int seed = 1)
+        {
+            var rng = new Random(seed);
+            for (int i = 0; i < numExamples; i++)
+            {
+                // Generate random, uncoupled features.
+                var data = new Data {
+                    Features = new float[2] { (float)(rng.NextDouble() - 0.5), (float)(rng.NextDouble() - 0.5) }
```

Comment:
>(float)(rng.NextDouble() - 0.5) [](start = 46, length = 31)

i would define this as a lambda below rng definition to make the code cleaner.

var rndFloat => (float)(rng.NextDouble() - 0.5)